lib/compilers/kotlin.js

```diff
@@ -80,6 +77,44 @@ export class KotlinCompiler extends JavaCompiler {
         ];
     }
 
+    /**
+     * Handle Kotlin execution.
+     *
+     * Kotlin execution differs in the way that Kotlin requires its standard
+     * standard library because that's where the runtime libraries such as
+     * kotlin.jvm.internal.Intrinsics is.
+     *
+     * Therefore, we append the -include-runtime and -d flags to specify where
+     * to output the jarfile which we will run using `java -jar`
+     *
+     * TODO(supergrecko): Find a better fix than this bandaid for execution
+     */
+    async handleInterpreting(key, executeParameters) {
+        const alteredKey = {
+            ...key,
+            options: ['-include-runtime', '-d', 'example.jar'],
+        };
+        const compileResult = await this.getOrBuildExecutable(alteredKey);
+        executeParameters.args = [
+            '-Xss136K', // Reduce thread stack size
+            '-XX:CICompilerCount=2', // Reduce JIT compilation threads. 2 is minimum
+            '-XX:-UseDynamicNumberOfCompilerThreads',
+            '-XX:-UseDynamicNumberOfGCThreads',
+            '-XX:+UseSerialGC', // Disable parallell/concurrent garbage collector
+            '-cp',
+            compileResult.dirPath,
+            '-jar',
+            'example.jar',
```

Comment:
Here we are basically saying oh we're adding three arguments (to pack it into a .jar) and executing that jar instead of the base JavaCompiler's .classfile execution

Reply:
i assume this still will work with multiple files if you pass them as arguments, right?

Reply:
It should, because the change only tells Kotlin where (and which format) to output the compilation result into.